### PR TITLE
Add sequence birecursive instance and fix corecursive tests

### PR DESCRIPTION
**Pull request by 1Jajen1**
I forgot this instance, also corecursive equality of unfolds by == instead of using Eq, that only fails with Sequence, but should be fixed by using equalUnderTheLaw anyway. (The same is not necessary for Recursive because we are only comparing ints there)

### CODE REVIEW

**Comment by 1Jajen1** [comment]
I also moved elgot and coelgot, it was a mistake to have them in recursive/corecursive.

File: modules/recursion/arrow-recursion-data/src/test/kotlin/arrow/recursion/SequenceKBirecursive.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package arrow.recursion
```
Comment (pakoito) [inline modules/recursion/arrow-recursion-data/src/test/kotlin/arrow/recursion/SequenceKBirecursive.kt]: Add Test to the name of this file, if possible :D

**Comment by 1Jajen1** [comment]
> LGTM, some nit on file names

Should be done :+1:

**Comment by pakoito** [comment]
All ABOOOOOARD!

**Pull request merged**